src/java/org/apache/cassandra/auth/CassandraCIDRAuthorizer.java

```diff
@@ -154,4 +154,10 @@ public boolean hasAccessFromIp(RoleResource role, InetAddress ipAddress)
                                                           TimeUnit.NANOSECONDS);
         return hasAccess;
     }
+
+    @VisibleForTesting
+    public boolean has(RoleResource role)
```

Comment:
Not sure whether this is the best place for this method, but I did not find any better so far. Open for ideas

Reply:
CassandraCIDRAuthorizer has two caches in it - CIDRPermissionsCache and CIDRGroupsMappingCache. Naming this function has() doesn't indicate which cache are we looking into. Also, the same function won't be available when AllowAllCIDRAuthorizer is used. I prefer moving this has() into CIDRPermissionsManager, so we can make this function available regardless of CIDR authorizer selected, also available for nodetool commands through CIDRPermissionsManager mbean. I mean similar to the way CIDRPermissionsManager::invalidateCidrPermissionsCache implemented. If you don't mind, I can make those changes and send them to you to include here.

Reply:
SGTM, thanks :-)

Reply:
We wouldn't need this has() function if we plan to not check for entry existence in the cache in unit tests to avoid flaky test. I will wait for your confirmation on the final approach before making changes.

Reply:
I am fine if we do not use that method, but how do we confirm the validation? The tests are passing with/without the fix. I was looking to cover the invalidation fully and ensure that we do not regress in the future.

Reply:
I am finding a way to control cache expiration in unit tests, either using ticker or larger cache validity value. Please give me some time, I will let you know my findings

Reply:
Makes sense, thanks for looking into that

Reply:
Mentioned a possible solution here https://github.com/apache/cassandra/pull/2648/files#r1332985152, will run through circle CI and will confirm.